chia/wallet/vc_wallet/vc_store.py

```diff
@@ -0,0 +1,239 @@
+from __future__ import annotations
+
+import dataclasses
+from functools import cmp_to_key
+from typing import Dict, List, Optional, Tuple, Type, TypeVar
+
+from aiosqlite import Row
+
+from chia.types.blockchain_format.coin import Coin
+from chia.types.blockchain_format.program import Program
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.util.db_wrapper import DBWrapper2
+from chia.util.ints import uint32, uint64
+from chia.util.streamable import Streamable, streamable
+from chia.wallet.lineage_proof import LineageProof
+from chia.wallet.util.merkle_utils import list_to_binary_tree
+from chia.wallet.vc_wallet.vc_drivers import VCLineageProof, VerifiedCredential
+
+
+@dataclasses.dataclass(frozen=True)
+class VCProofs:
+    key_value_pairs: Dict[str, str]
+
+    def as_program(self) -> Program:
+        def byte_sort_pairs(f1: Tuple[str, str], f2: Tuple[str, str]) -> int:
+            return 1 if Program.to([10, (1, f1[0]), (1, f2[0])]).run([]) == Program.to(None) else -1
+
+        prog: Program = Program.to(
+            list_to_binary_tree(
+                list(
+                    sorted(
+                        self.key_value_pairs.items(),
+                        key=cmp_to_key(byte_sort_pairs),
+                    )
+                )
+            )
+        )
+        return prog
+
+    def root(self) -> bytes32:
+        return self.as_program().get_tree_hash()
+
+    @staticmethod
+    def from_program(prog: Program) -> VCProofs:
+        first: Program = prog.at("f")
+        rest: Program = prog.at("r")
+        if first.atom is None and rest.atom is None:
+            final_dict: Dict[str, str] = {}
+            final_dict.update(VCProofs.from_program(first).key_value_pairs)
+            final_dict.update(VCProofs.from_program(rest).key_value_pairs)
+            return VCProofs(final_dict)
+        elif first.atom is not None and rest.atom is not None:
+            return VCProofs({first.atom.decode("utf-8"): rest.atom.decode("utf-8")})
+        else:
+            raise ValueError("Malformatted VCProofs program")  # pragma: no cover
+
+
+_T_VCStore = TypeVar("_T_VCStore", bound="VCStore")
+
+
+@streamable
+@dataclasses.dataclass(frozen=True)
+class VCRecord(Streamable):
+    vc: VerifiedCredential
+    confirmed_at_height: uint32  # 0 == pending confirmation
+
+
+def _row_to_vc_record(row: Row) -> VCRecord:
+    return VCRecord(
+        VerifiedCredential(
+            Coin(bytes32.from_hexstr(row[2]), bytes32.from_hexstr(row[3]), uint64.from_bytes(row[4])),
+            LineageProof.from_bytes(row[5]),
+            VCLineageProof.from_bytes(row[6]),
+            bytes32.from_hexstr(row[0]),
+            bytes32.from_hexstr(row[7]),
+            bytes32.from_hexstr(row[8]),
+            None if row[9] is None else bytes32.from_hexstr(row[9]),
+        ),
+        uint32(row[10]),
+    )
+
+
+class VCStore:
```

Comment:
@arvidn noticed we're missing unit tests for these which I missed.
We need unit tests for:
 - VCStore
 - VCProofs
 - VerifiedCredential